pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -216,8 +218,17 @@
     private Channel listenChannel;
     private Channel listenChannelTls;
 
+    private final long maxMessagePublishBufferSize;
+    private final long resumeProducerReadMessagePublishBufferSize;
+    private volatile long currentMessagePublishBufferSize;
```

Comment:
```suggestion
    private volatile long currentMessagePublishBufferBytes;
```

I prefer using `bytes` rather than `size` to make the unit more explicit.